superset-frontend/src/views/CRUD/data/dataset/AddDataset/index.tsx

```diff
@@ -73,6 +75,38 @@ export default function AddDataset() {
     Reducer<Partial<DatasetObject> | null, DSReducerActionType>
   >(datasetReducer, null);
   const [hasColumns, setHasColumns] = useState(false);
+  const [linkedDatasets, setLinkedDatasets] = useState<DatasetObject[]>([]);
+  const linkedDatasetNames = linkedDatasets.map(dataset => dataset.table_name);
+  const encodedSchema = dataset?.schema
+    ? encodeURIComponent(dataset?.schema)
+    : undefined;
+
+  const queryParams = dataset?.schema
+    ? rison.encode_uri({
+        filters: [
+          { col: 'schema', opr: 'eq', value: encodedSchema },
+          { col: 'sql', opr: 'dataset_is_null_or_empty', value: '!t' },
+        ],
+      })
+    : undefined;
+
+  const getDatasetsList = () => {
```

Comment:
I'd recommend moving any api calls outside of the view component into either a hook or redux action for separation of concerns.

Reply:
Moved the API call into a hook in [`this commit`](https://github.com/apache/superset/pull/22136/commits/ad13ff4dbb6c72d647397d3b2d106c6fc3dbbe2c).